.github/workflows/rust-pipeline.yml

```diff
@@ -0,0 +1,101 @@
+name: Rust pipeline
+
+on:
+  workflow_call:
+    inputs:
+      runs-on:
+        required: false
+        type: string
+        default: ubuntu-latest-16core
+
+# Environment variables for all jobs.
+env:
+  CARGO_TERM_COLOR: always
+  ADDITIONAL_PACKAGES: ""
+
+# Defined CI jobs.
+jobs:
+  check:
+    runs-on: ${{ inputs.runs-on }}
+    container: ghcr.io/famedly/rust-container:nightly
+    steps:
+    - name: Checkout current repository
+      uses: actions/checkout@v3
+      with:
+        path: head
+
+    - uses: famedly/backend-build-workflows/.github/actions/rust-prepare@main
+      with:
+        gitlab_ssh: ${{ secrets.CI_SSH_PRIVATE_KEY}}
+        gitlab_user: ${{ secrets.GITLAB_USER }}
+        gitlab_pass: ${{ secrets.GITLAB_PASS }}
+    
+    - name: Caching
+      uses: Swatinem/rust-cache@b8a6852b4f997182bdea832df3f9e153038b5191
+      with:
+        workspaces: "head -> target"
```

Comment:
Should we maybe put both workspace into the same target folder so that both workspaces can use the compiled dependencies? Would that work? (with .cargo config setting the target dir)

Reply:
Thinking about it again, I don't think that it's worth it
- we cache lcov
- I'd ideally need to keep the cache key based on combination of parent branch and current branch
- dunno. Maybe we can try in future, but the benefit won't be large imo